Fix #1721 - CreateTable generation for DateTime2 with Precision.

Corey-M opened the pull request:
DateTime2 type first available in MS SQL 2008.  Added override to `SqlServer2008SqlBuilder` to correctly generate type string for DateTime2 with Precision.

Test add but not tested, fix confirmed in external program.  Please check this test.

Corey-M commented:
Fixes #1721

Corey-M commented:
OK, I made a bad assumption without checking, assumed that the various `SqlServerXXXXSqlBuilder` versions built off each other.  I'll shift the fix into the root `SqlServerSqlBuilder` class.

sdanyliv commented on Tests/Linq/UserTests/Issue1721Tests.cs:
```diff
@@ -0,0 +1,32 @@
+			public DateTime BadField { get; set; }
+		}
+
+		[ActiveIssue(1721, Configuration = ProviderName.SqlServer2008)]
```
ActiveIssue is not needed if bug fixed.

Corey-M replied:
I was copying from another test.  Will fix.

Corey-M commented:
Hopefully this is the last change.  It's not super important but not specifying the precision defaults to 7 for `DateTimeOffset`, `DateTime2` and `Time` types if you don't specify.  This should properly handle all the things.

sdanyliv commented on Tests/Linq/UserTests/Issue1721Tests.cs:
```diff
@@ -0,0 +1,37 @@
+				Assert.DoesNotThrow(() =>
+				{
+					using (var temp = db.CreateTempTable<I1721Model>("Issue1721"))
+					{ }
```
I think we have to check here that `LastQuery` contains DataType(Precision) text for every type.

Corey-M replied:
You're right.  And while working on this I found an error that the test runner didn't throw.

Corey-M commented:
Previous code was falling through to the default code when `Precision` is specified as the default value of `7` for all types.  Fixed that and added testing of generated SQL to ensure that the right values are created.
